lib/puppet/provider/service/smf.rb

```diff
@@ -49,6 +49,52 @@ def self.instances
    end
   end
 
+  # Returns the service's FMRI. We fail if multiple FMRIs correspond to
+  # @resource[:name].
+  #
+  # If the service does not exist or we fail to get any FMRIs from svcs,
+  # this method will raise a Puppet::ExecutionFailure
```

Comment:
Down at line 68 (and line 83), we're raising a Puppet::Error, not a Puppet::ExecutionFailure. Is this what was intended?

Reply:
Puppet::Error is the right choice, I'll fix that.

Reply:
Oh, I updated the PR to fix this.